micro_sam/training/semantic_sam_trainer.py

```diff
@@ -13,31 +39,34 @@ class SemanticSamTrainer(DefaultTrainer):
     def __init__(
         self,
         convert_inputs,
-        num_classes: int = 1,
+        num_classes: int,
         **kwargs
     ):
-        loss = DiceLoss()
-        metric = DiceLoss()
+        assert num_classes > 1
+
+        loss = CustomDiceLoss(num_classes=num_classes)
+        metric = CustomDiceLoss(num_classes=num_classes)
         super().__init__(loss=loss, metric=metric, **kwargs)
```

Comment:
Let's pass the `SemanticSamLogger` for the logger class here.

Reply:
Done. Thanks!